### PR TITLE
Allow additional variables in inputSubmit and submitContainer templates

### PR DESCRIPTION
**Pull request by Holt59**
Currently, it is possible to [add additional variables to templates](http://book.cakephp.org/3.0/en/views/helpers/form.html#adding-additional-template-variables-to-templates) for all inputs and input containers **except** hidden inputs and submit inputs due to [this conditional](https://github.com/cakephp/cakephp/blob/master/src/View/Helper/FormHelper.php#L1021) statement.

I am suggesting a patch to support additional variables for the `inputSubmit` and `submitContainer`  templates (PR opened after suggestion in [this issue](https://github.com/cakephp/cakephp/issues/7958#issuecomment-168629979)).

### CODE REVIEW

**Comment by dereuromark** [comment]
A small test case would be nice to avoid regression.

File: src/View/Helper/FormHelper.php [hunk 1]
```diff
@@ -1722,7 +1722,11 @@ public function submit($caption = null, array $options = [])
         if (!is_string($caption) && empty($caption)) {
             $caption = __d('cake', 'Submit');
         }
-        $options += ['type' => 'submit', 'secure' => false];
+        $options += [
+            'type' => 'submit',
+            'secure' => false,
+            'templateVars' => []
```
Comment (markstory) [inline src/View/Helper/FormHelper.php]: This new option should be added to the `Options` section in the doc block.

Comment (Holt59) [inline src/View/Helper/FormHelper.php]: Option added. Maybe it is worth adding the options to the others method using it? It is currently only documented in the Cookbook.

**Comment by ADmad** [comment]
Why are people still using `<input>` instead of `<button>` for submitting ? :)

**Pull request merged**